src/test/java/de/retest/recheck/report/TestReportFilterTest.java

```diff
@@ -134,6 +134,27 @@ void element_difference_and_child_differences_should_be_filtered_properly() thro
 				.containsExactly( notFilterMe );
 	}
 
+	@Test
+	void filter_element_difference_should_have_no_differences_if_filtered() {
+		final AttributeDifference attributeDifference = mock( AttributeDifference.class );
+		final AttributesDifference attributes = mock( AttributesDifference.class );
+		when( attributes.getDifferences() ).thenReturn( Collections.singletonList( attributeDifference ) );
+
+		final ElementDifference difference = mock( ElementDifference.class );
+		when( difference.getAttributesDifference() ).thenReturn( attributes );
+
+		final Filter filterAll = mock( Filter.class );
+		when( filterAll.matches( any() ) ).thenReturn( true );
+		when( filterAll.matches( any(), any() ) ).thenReturn( true );
+
+		final ElementDifference filter = TestReportFilter.filter( difference, filterAll );
```

Comment:
`filter` is really confusing since it is an `ElementDifference`, not a filter. How about `filtered` or `filteredDifference`?